runsc/cmd/gofer.go

```diff
@@ -100,6 +103,15 @@ func (g *Gofer) Execute(_ context.Context, f *flag.FlagSet, args ...interface{})
 		return subcommands.ExitUsageError
 	}
 
+	if g.attached {
+		// Ensure this process is killed after parent process terminates when
+		// attached mode is enabled. In the unfortunate event that the parent
+		// terminates before this point, this process leaks.
+		if err := unix.Prctl(unix.PR_SET_PDEATHSIG, uintptr(unix.SIGKILL), 0, 0, 0); err != nil {
+			Fatalf("error setting parent death signal: %v", err)
+		}
+	}
+
```

Comment:
Why do you need this after setting SysProcAttr.Pdeathsig on process creation?

Reply:
Looks like I don't. I thought we were losing the death signal on the re-exec to drop caps (like boot does), but it looks like it actually sticks around since there is no uid change or _increase_ in caps.

Removed.